pkg/service/restore_data.go

```diff
@@ -89,6 +96,25 @@ func (r *dataRestorer) Restore(request *model.RestoreRequest) (model.RestoreJobI
 	return jobID, nil
 }
 
+// validateDestinationNamespace checks if destination cluster contains namespace from restore request (if it is set).
+func (r *dataRestorer) validateDestinationNamespace(request *model.RestoreRequest, jobID model.RestoreJobID) bool {
+	if request.Policy.Namespace != nil {
+		missingNamespaces := r.nsValidator.MissingNamespaces(
+			request.DestinationCuster, []string{*request.Policy.Namespace.Destination})
+		if len(missingNamespaces) > 0 {
+			// it can be only 1 missing ns.
+			err := fmt.Errorf("destination cluster does not have namespace %s", missingNamespaces[0])
+			slog.Error("Failed to restore by path",
+				slog.Any("cluster", request.DestinationCuster),
```

Comment:
This logging may leak credentials.

Reply:
* `request.DestinationCuster` is a pointer.
* DestinationCuster -> DestinationCluster